pytest_invenio/fixtures.py

```diff
@@ -947,3 +947,41 @@ def test_with_user(service, myuser):
 
     """
     return UserFixtureBase
+
+
+@pytest.fixture(scope="function")
+def set_tmp_config(base_app):
+    """Fixture to temporarily set config values.
+
+    Oftentimes, tests set config values but don't first save them and later
+    (after test concluded) restore them. This causes test leakage because
+    base_app (the Flask application) is a module scoped fixture.
+    `set_tmp_config` provides a function to call to temporarily set config
+    values automatically: without further intervention, it resets them to their
+    original values after the test is done.
+
+    Scope: function
+
+    .. code-block:: python
+
+        def test_with_tmp_config(app, set_tmp_config):
+            set_tmp_config({
+                "MAX_CONTENT_LENGTH": 2**20,
+                "SECRET_KEY": "foo"
+            })
+            # app.config will contain MAX_CONTENT_LENGTH = 2**20 and
+            # SECRET_KEY = "foo" .
+            # No other call is needed to revert values.
+            # Other tests will see the original values for these configs.
+    """
+    config_prev = {}
+
+    def _set_tmp_config(config):
+        for key, value in config.items():
+            config_prev[key] = base_app.config[key]
```

Comment:
Naive question: what if the key is not set previously?

Reply:
Addressed!